ds3-autogen-c/src/test/java/com/spectralogic/ds3autogen/c/StructHelper_Test.java

```diff
@@ -264,4 +267,37 @@ public void testHasAttributesMixed() {
         assertTrue(StructHelper.hasAttributes(testAttributesStruct));
         assertTrue(StructHelper.hasChildNodes(testAttributesStruct));
     }
+
+    @Test
+    public void testOrderingStructsWorks() throws ParseException {
+        final Struct testOrderInnerStruct = new Struct("testOrderInnerStruct",
+                ImmutableList.of(
+                        new StructMember(new FreeableType("ds3_str", true), "name", "name", null, false, false)));
+        final Struct testOrderOutterStruct = new Struct("testOrderOuterStruct",
+                ImmutableList.of(
+                        new StructMember(new FreeableType("testOrderInnerStruct", true), "objects_node", "objects_node", null, false, false)));
+        final ImmutableList<Struct> structsList = ImmutableList.of(testOrderOutterStruct, testOrderInnerStruct);
+        final ImmutableList<Struct> orderedStructsList = StructHelper.getStructsOrderedList(structsList, ImmutableSet.of());
+
+        assertThat(orderedStructsList.indexOf(testOrderInnerStruct), is(0));
+    }
+
+    @Test
```

Comment:
You can simplify this test by changing this to:
@Test (expected = ParseException.class)

then you don't have to use a try-catch block, and the test will just terminate when the exception is thrown